database/trackPlays.ts

```diff
@@ -1,9 +1,34 @@
+import { endOfDay, startOfDay } from "date-fns";
+import { FilterQuery } from "mongoose";
+
 import testDatabase from "../tests/database/testDatabase";
+import { ITrackPlay, TrackPlay } from "../models/TrackPlay";
+
+interface GetTrackPlaysParameters {
+  startDate?: Date;
+  endDate?: Date;
+}
+
+const getFilterQuery = (requestParameters?: GetTrackPlaysParameters) => {
+  let filter: FilterQuery<ITrackPlay> = {};
+
+  if (requestParameters?.startDate || requestParameters?.endDate) {
+    filter = { endTime: {} };
+    if (requestParameters.startDate) {
+      filter.endTime.$gte = startOfDay(new Date(requestParameters.startDate));
+    }
+    if (requestParameters.endDate) {
+      filter.endTime.$lte = endOfDay(new Date(requestParameters.endDate));
+    }
+  }
+
+  return filter;
+};
 
-import { TrackPlay } from "../models/TrackPlay";
+const getTrackPlays = async (
+  parameters?: GetTrackPlaysParameters
+): Promise<ITrackPlay[]> => TrackPlay.find(getFilterQuery(parameters));
 
-const getTrackPlays = async () => TrackPlay.find({});
+const database = { getTrackPlays };
 
-export default process.env.NODE_ENV === "test"
-  ? testDatabase
-  : { getTrackPlays };
+export default process.env.NODE_ENV === "test" ? testDatabase : database;
```

Comment:
I can export only `getTrackPlays` here and in other file I would import this `getTrackPlays` and then export the right database (it'll depend on environment)